portal/src/util/formatDatetime.ts

```diff
@@ -0,0 +1,16 @@
+import { DateTime } from "luxon";
+
+export function formatDatetime(
+  isoTimeString: string | null,
+  locale?: string,
+  format: string = "dd/MM/yyyy HH:mm"
```

Comment:
1. The argument `locale` should be the first argument. The reason is that it is more friendly to function currying (though javascript function are not curried automatically, but this is a common practice in functional programming)
2. Instead of a hard-coded format, we should use predefined format. In https://moment.github.io/luxon/docs/manual/formatting.html we should use `DATETIME_SHORT`

So this function should look like

```typescript
function formatDate(locale: string, date: Date | string | null): string | null {
  let datetime;
  if (date instanceof Date) {
    datetime = DateTime.fromJSDate(date);
  } else if (typeof date === "string") {
    datetime = DateTime.fromISO(date);
  }

  if (datetime == null) {
    return null;
  }

  datetime = datetime.setLocale(locale);
  return datetime.toLocaleString(DateTime.DATETIME_SHORT);
}
```

Reply:
> more friendly to function currying

So you mean it is more convenient to define a version for specific locale by reducing it like currying in functional programming?

Reply:
```typescript
const formatDateInEnglish = formatDate.bind("en");
const formatDateInChinese = formatDate.bind("zh");
```